### ISSUE TITLE
Support different providers

### ISSUE THREAD

**Issue by anbraten** [nuxt-hub/core#9]
In addition to cloudflare support for different providers like:

- [Vercel](https://vercel.com/)
- [Netlify](https://www.netlify.com/)
- [Lagoss](https://docs.lagoss.com)

would be great.

**Comment by marcushohlbein** [comment]
Vercel has it's own packages for this

- [https://vercel.com/docs/storage](https://vercel.com/docs/storage)

**Comment by Atinux** [comment]
Indeed, so far it is quite easy to add storage for a Nuxt application using Vercel.

**Comment by rexhent** [comment]
> In addition to cloudflare support for different providers like:
> 
>     * [Vercel](https://vercel.com/)
> 
>     * [Netlify](https://www.netlify.com/)
> 
>     * [Lagoss](https://docs.lagoss.com)
> 
> 
> would be great.

Maybe AWS Amplify and Azure eventually?

I don't know much about nuxthub's architecture, but is self hosting something in the realm of possibility? In a way that is worth the time? Or is it possible to intergrate with something like Coolify?

**Comment by LeonardoRick** [comment]
Is that at least possible to bypass nuxthub when deploying to other providers? I came across this thread because I was testing my project deployment in multiple providers but them once adding `@nuxthub/core` as a plugin of my project I can't deploy on vercel or netlify anymore. 

|Vercel output|
|--|
|![image](https://github.com/user-attachments/assets/a0300484-ab3a-4ec6-8891-224bd55c6b75)|

I want to add all the nuxt hub setup on my project but also be able to deploy it optimized for other providers as well. Is that possible?
